File: lib/src/common/pipes.dart
/**
 * @module
 * @description
 * This module provides a set of common Pipes.
 */
library angular2.src.common.pipes;

import "pipes/async_pipe.dart" show AsyncPipe;
import "pipes/uppercase_pipe.dart" show UpperCasePipe;
import "pipes/lowercase_pipe.dart" show LowerCasePipe;
import "pipes/json_pipe.dart" show JsonPipe;
import "pipes/slice_pipe.dart" show SlicePipe;
import "pipes/date_pipe.dart" show DatePipe;
import "pipes/number_pipe.dart" show DecimalPipe, PercentPipe, CurrencyPipe;
export "pipes/async_pipe.dart" show AsyncPipe;
export "pipes/date_pipe.dart" show DatePipe;
export "pipes/json_pipe.dart" show JsonPipe;
export "pipes/slice_pipe.dart" show SlicePipe;
export "pipes/lowercase_pipe.dart" show LowerCasePipe;
export "pipes/number_pipe.dart"
    show NumberPipe, DecimalPipe, PercentPipe, CurrencyPipe;
export "pipes/uppercase_pipe.dart" show UpperCasePipe;

/**
 * A collection of Angular core pipes that are likely to be used in each and every
 * application.
 *
 * This collection can be used to quickly enumerate all the built-in pipes in the `pipes`
 * property of the `@Component` or `@View` decorators.
 */
const COMMON_PIPES = const [
  AsyncPipe,
  UpperCasePipe,
  LowerCasePipe,
  JsonPipe,
  SlicePipe,
  DecimalPipe,
  PercentPipe,
  CurrencyPipe,
  DatePipe
];

File: modules/angular2/src/common/pipes.ts
/**
 * @module
 * @description
 * This module provides a set of common Pipes.
 */
import {AsyncPipe} from './pipes/async_pipe';
import {UpperCasePipe} from './pipes/uppercase_pipe';
import {LowerCasePipe} from './pipes/lowercase_pipe';
import {JsonPipe} from './pipes/json_pipe';
import {SlicePipe} from './pipes/slice_pipe';
import {DatePipe} from './pipes/date_pipe';
import {DecimalPipe, PercentPipe, CurrencyPipe} from './pipes/number_pipe';
import {ReplacePipe} from './pipes/replace_pipe';
import {CONST_EXPR} from 'angular2/src/facade/lang';

export {AsyncPipe} from './pipes/async_pipe';
export {DatePipe} from './pipes/date_pipe';
export {JsonPipe} from './pipes/json_pipe';
export {SlicePipe} from './pipes/slice_pipe';
export {LowerCasePipe} from './pipes/lowercase_pipe';
export {NumberPipe, DecimalPipe, PercentPipe, CurrencyPipe} from './pipes/number_pipe';
export {UpperCasePipe} from './pipes/uppercase_pipe';
export {ReplacePipe} from './pipes/replace_pipe';

/**
 * A collection of Angular core pipes that are likely to be used in each and every
 * application.
 *
 * This collection can be used to quickly enumerate all the built-in pipes in the `pipes`
 * property of the `@Component` or `@View` decorators.
 */
export const COMMON_PIPES = CONST_EXPR([
  AsyncPipe,
  UpperCasePipe,
  LowerCasePipe,
  JsonPipe,
  SlicePipe,
  DecimalPipe,
  PercentPipe,
  CurrencyPipe,
  DatePipe,
  ReplacePipe
]);
